File: DemoServer/AdditionalLanguages/python/staticIndexes/mapReduceIndex/mapReduceIndex.py
from demo_example import Example, RunParamsBase

# region Usings
from ravendb import AbstractIndexCreationTask
#endregion

class RunParams(RunParamsBase):
    def __init__(self, country: str):
        self.country = country

#region Demo
#region Step_1
class Employees_ByCountry(AbstractIndexCreationTask):
#endregion

    #region Step_2
    class IndexEntry:
        def __init__(self, country: str, country_count: int):
            self.Country = country
            self.CountryCount = country_count
    #endregion

    def __init__(self):
        #region Step_3
        super().__init__()
        self.map = (
            "docs.Employees.Select(employee => new { "
            "    Country = employee.Address.Country, "
            "    CountryCount = 1 "
            "})"
        )
        #endregion
        #region Step_4
        self.reduce = (
            "results.GroupBy(result => result.Country).Select(g => new { "
            "    Country = g.Key, "
            "    CountryCount = Enumerable.Sum(g, x => x.CountryCount) "
            "})"
        )
        #endregion
#endregion

class MapReduceIndex(Example):
    def run(self, run_params: RunParams) -> int:
        country = run_params.country
        Employees_ByCountry().execute(self.document_store_holder.store())
        #region Demo
        with self.document_store_holder.store().open_session() as session:
            #region Step_5
            query_result = (
                session.query_index_type(Employees_ByCountry, Employees_ByCountry.IndexEntry)
                .where_equals("Country", country)
                .single()
            )
            number_of_employees_in_country = query_result.CountryCount if query_result is not None else 0
            #endregion
        #endregion

        return number_of_employees_in_country

File: DemoServer/AdditionalLanguages/php/staticIndexes/mapReduceIndex/MapReduceIndex.php
<?php

namespace RavenDB\Demo\staticIndexes\mapReduceIndex;

//region Usings
use RavenDB\Documents\Indexes\AbstractIndexCreationTask;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;

//region Demo
//region Step_1
class Employees_ByCountry extends AbstractIndexCreationTask
//endregion
{
    public function __construct()
    {
        //region Step_2
        parent::__construct();

        $this->map = "docs.Employees.Select(employee => new { " .
            "    Country = employee.Address.Country, " .
            "    CountryCount = 1 " .
            "})";
        //endregion

        //region Step_3
        $this->reduce = "results.GroupBy(result => result.Country).Select(g => new { " .
            "    Country = g.Key, " .
            "    CountryCount = Enumerable.Sum(g, x => x.CountryCount) " .
            "})";
        //endregion
    }
}
//endregion

//region Demo
//region Step_4
class Employees_ByCountryResult
{
    private ?String $country = null;
    private ?int $countryCount = null;

    public function getCountry(): ?string
    {
        return $this->country;
    }

    public function setCountry(?string $country): void
    {
        $this->country = $country;
    }

    public function getCountryCount(): ?int
    {
        return $this->countryCount;
    }

    public function setCountryCount(?int $countryCount): void
    {
        $this->countryCount = $countryCount;
    }
}
//endregion
//endregion

class MapReduceIndex
{
    public function __invoke(RunParams $runParams)
    {
        $country = $runParams->getCountry();
        $numberOfEmployeesFromCountry = 0;

        //region Demo
        $session = DocumentStoreHolder::getStore()->openSession();
        try {
            //region Step_5
            $queryResult = $session->query(Employees_ByCountryResult::class, Employees_ByCountry::class)
                ->whereEquals("Country", $country)
                ->firstOrDefault();

            $numberOfEmployeesFromCountry = $queryResult != null ? $queryResult->getCountryCount() : 0;
            //endregion
        } finally {
            $session->close();
        }
        //endregion

        return $numberOfEmployeesFromCountry;
    }
}
